src/Project.php

```diff
@@ -2669,6 +2669,200 @@ public function showDebug()
     }
 
+    /**
+     * Get the list of projects for a user.
+     *
+     * @param int $user_id The user ID.
+     * @param bool $search_in_groups Whether to search in groups.
+     * @param bool $search_in_team Whether to search in the team.
+     * @return array The list of project IDs.
+     */
+    public static function getProjectIDsForUser(
+        int $user_id,
+        bool $search_in_groups = true,
+        bool $search_in_team = true
+    ): array {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $req = [
+            'FROM' => Project::getTable(),
+            'WHERE' => [
+                'OR' => [
+                    'users_id' => $user_id
+                ]
+            ]
+        ];
+
+        $groups_sub_query = new \Glpi\DBAL\QuerySubQuery([
+            'SELECT' => [
+                'groups_id'
+            ],
+            'FROM' => Group_User::getTable(),
+            'WHERE' => [
+                'users_id' => $user_id
+            ]
+        ]);
+
+        if ($search_in_groups) {
+            $req['WHERE']['OR']['groups_id'] = $groups_sub_query;
+        }
+
+        if ($search_in_team) {
+            $crit = [
+                ['itemtype' => 'User', 'items_id' => $user_id]
+            ];
+
+            if ($search_in_groups) {
+                $crit[] = ['itemtype' => 'Group', 'items_id' => $groups_sub_query];
+            }
+
+            $team_sub_query = new \Glpi\DBAL\QuerySubQuery([
+                'SELECT' => [
+                    'projects_id'
+                ],
+                'FROM' => ProjectTeam::getTable(),
+                'WHERE' => [
+                    'OR' => $crit
+                ]
+            ]);
+
+            $req['WHERE']['OR']['id'] = $team_sub_query;
+        }
+
+        $iterator = $DB->request($req);
+
+        $project_ids = [];
+        foreach ($iterator as $data) {
+            $project_ids[] = $data['id'];
+        }
+
+        return $project_ids;
+    }
+
+    /**
+     *  Show the list of projects for a user in the personal view.
+     *
+     * @param int $user_id The user ID.
+     * @return void
+     */
+    public static function showListForCentral(int $user_id): void
+    {
+        $projects = array_map(
+            fn($id) => self::getById($id),
+            self::getProjectIDsForUser($user_id, false, true)
+        );
+
+        // If no project are found, do not display anything
+        if (empty($projects)) {
+            return;
+        }
+
+        // If all project are finished, do not display anything
+        if (
+            empty(array_filter($projects, function ($project) {
+                $state = ProjectState::getById($project->fields['projectstates_id']);
+                if ($state !== false) {
+                    if ($state->fields['is_finished']) {
+                        return false;
+                    }
+                }
+
+                return true;
+            }))
+        ) {
+            return;
+        }
```

Comment:
Since `getProjectIDsForUser` is only used here it couldn't it directly return the non finished projects ?
This can be done by adding an `active` parameter or renaming the method to `getActiveProjectIDsForUser`.

This would lead to better performances by getting the relevant projects ids with a single SQL request, without the need to iterate on its data and load a State item for each entry.